apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -416,6 +449,38 @@ impl Telemetry {
         });
         futures::executor::block_on(jh).expect("failed to replace tracer provider");
     }
+
+    fn router_service_span(config: apollo::Config) -> impl Fn(&RouterRequest) -> Span + Clone {
+        let client_name_header = config.client_name_header;
+        let client_version_header = config.client_version_header;
+
+        move |request: &RouterRequest| {
+            let http_request = &request.context.request;
+            let headers = http_request.headers();
+            let query = http_request.body().query.clone().unwrap_or_default();
+            let operation_name = http_request
+                .body()
+                .operation_name
+                .clone()
+                .unwrap_or_else(|| "".to_string());
```

Comment:
Check changes here to have the same behavior https://github.com/apollographql/router/pull/869/files

Reply:
I was wondering. By the time that we get here the request should have either been hydrated from the post data or the url. If it hasn't then get would never work.
WDYT?

Reply:
It changed, because before with warp, if we had a GET request and we asked for the body, it worked. Now with axum, it's stricter, and we are taking the original request, to be sure if someone wants to detect a very specific pattern in a plugin regarding the method, the url, and so one then it's now possible. So I think it won't work at all if you do not update this part of code.

Reply:
I feel like the reason why it's the case is we're missusing axum extractors, IE we should have very narrow extracts instead of grabbing the whole HTTP request. It however makes sense given the nature of the software we're building ^^

Reply:
I kinda disagree :p because even if we are using different extractors, like extract for headers, for body, for query, url and so one. At the end we're rebuilding the exactly same request (except for the URL). Then for me it makes sense to just fetch the original request, and pass it over all our plugins. It's less error prone and we just pass the original request.

Reply:
So just to be clear on current functionality, if a GET request comes in the graphql::Request is never populated?

Reply:
I do see the point that the originating request should exactly match the originating request, but there's something deeper here that needs to be discussed.

Reply:
@bnjjj I've migrated the code from the Axum http factory and we can follow up on the course of action for get requests in a separate ticket.

Reply:
Do https://github.com/apollographql/router/pull/877 and https://github.com/apollographql/router/pull/873 remove the need for a followup discussion ? or shall we open a followup issue ?

Reply:
Yeah, I think this is moot now. The operationName is always present in the request.